### PR TITLE
feat: add logic for maintenance page and database sync

### PR DESCRIPTION
**Pull request by junminahn**
- display the maintenance page until Golddr Keycloak is up and ready.
- add logic to compare active cluster's xlog and standby cluster's xlog for database sync.

### CODE REVIEW

File: transition-scripts/deploy.sh [hunk 1]
```diff
@@ -26,18 +26,30 @@ if [ "$#" -lt 1 ]; then
 fi
 
 namespace=$1
+purge="false"
 
 pwd="$(dirname "$0")"
 source "$pwd/helpers/_all.sh"
 
-# TODO: take an argument to just upgrade; no uninstallation, and no cleanup to ensure gold is running as active and golddr is running as standby
+while [[ "$2" =~ ^- && ! "$2" == "--" ]]; do
```
Comment (thegentlemanphysicist) [inline transition-scripts/deploy.sh:34]: I'd just like to confirm my understanding of the logic here:
The first check seems to ensure the $2 argument script starts with a dash, but isn't "--", 
The following logic checks if the the $2 argument is "--purge"  or "-p"
The shift command lets this loop run from the 3rd argument until it hits an arg that does not start with a '-'.  Is that correct?

File: transition-scripts/helpers/helm.sh [hunk 2]
```diff
@@ -14,8 +14,10 @@ upgrade_helm() {
   cluster_mode="$2" # active | standby
   current=$(get_current_cluster)
 
-  helm repo add sso-charts https://bcgov.github.io/sso-helm-charts
-  helm repo update
+  {
+    helm repo add sso-charts https://bcgov.github.io/sso-helm-charts
+    helm repo update
+  } &>/dev/null
```
Comment (thegentlemanphysicist) [inline transition-scripts/helpers/helm.sh:20]: why are we suppressing the output here?  Jus to keep the github action cleaner?

File: transition-scripts/helpers/patroni.sh [hunk 3]
```diff
@@ -135,3 +139,72 @@ wait_for_patroni_all_ready() {
...
   while wait_ready; do sleep 5; done
 }
+
+get_patroni_xlog() {
+  if [ "$#" -lt 1 ]; then exit 1; fi
+
+  namespace="$1"
+  patroni_mode=$(check_patroni_cluster_mode "$namespace")
+  if [ "$patroni_mode" == "active" ]; then
+    leader=$(kubectl -n "$namespace" exec sso-patroni-0 -- patronictl list -f json | jq -r '.[] | select(.Role == "Leader") | .Member')
```
[the remark this answers is not in the document]
Comment (junminahn) [reply transition-scripts/helpers/patroni.sh:149]: yes, only the leader pod has the information of the most recent `xlog` size, and others hold the information of the individual's `received xlog` size.

File: transition-scripts/deploy.sh (hunk 1 above)
Comment (thegentlemanphysicist) [reply transition-scripts/deploy.sh:34]: excellent thank you

**Pull request merged**